pyiron_base/jobs/job/generic.py

```diff
@@ -653,16 +653,19 @@ def remove_child(self):
         _kill_child(job=self)
         super(GenericJob, self).remove_child()
 
-    def kill(self):
-        if self.status.running or self.status.submitted:
-            master_id, parent_id = self.master_id, self.parent_id
-            self.remove()
-            self.reset_job_id()
-            self.master_id, self.parent_id = master_id, parent_id
```

Comment:
Where do you handle the `job_id` reset? It would be great to maintain the old functionality in addition to the new functionality.

Reply:
Why is this necessary? I don't quite understand the code. 

```
    def reset_job_id(self, job_id=None):
        """
        Reset the job id sets the job_id to None in the GenericJob as well as all connected modules like JobStatus.
        """
        if job_id is not None:
            job_id = int(job_id)
        self._job_id = job_id
        self._status = JobStatus(db=self.project.db, job_id=self._job_id)
```

The goal of this implementation of kill is to act exactly as if the run did happen, but the user intervenes, but all generated input and output thus far are preserved. What does resetting the job id achieve exactly in the old implementation?

Reply:
If you calculate an energy volume curve with the `Murnaghan` class and one of the calculation fails, then you want to be able to restart the calculation that failed. In the background this is using the existing/ old kill functionality.

Reply:
Now that I understand the use case for the old method, I would prefer that we can move the old function to it's own thing and let this new method be kill() as I feel quite strongly that this specific use does not warrant occupying this name.

If I understand the use correctly, the removal of the directory is necessary for the restart yes? Which is why the old kill method employed the `remove` method. This is not compatible with the new purpose of kill() and what it intends to achieve. 

Any comment? Can you link me the specific code in pyiron that uses this? Or is this something that you used in your own notebooks?

Reply:
**tl;dr** Functions which occupy obvious names such as `kill` for the job object should not have niche functionality which is unexpected and exists because of legacy reasons, or backwards compatibility pains.

Just to explain why I feel quite strongly about this, I am viewing this from the perspective of a new user or the general potential pyiron userbase.

I'm looking for functionality that kills the job in the queueing system, while preserving the job data generated so far. Extremely common operation in the terminal, or any HPC user. I should find something in pyiron, since it claims that the need to access the terminal when I use pyiron should be almost nothing. For something as obvious as killing non-converging jobs on the queueing system, something like `qdel` or `scancel` functionality, surely this should exist! To my delight, I find that `job.kill()` exists. If I'm careful, I check the docstring, and I'm not too convinced, but I can't be bothered to see if it actually does what it is claiming through careful inspection of the source, so I call it. 

To my dismay, I find that it removes the entire directory for whatever reason! WTH? This makes no sense to me! Why doesn't it just delete/kill the job via the queueing system commands, like I would expect? Why is this method named `kill` when it doesn't agree with any conventions in HPC terminology?

Then, I go peek into the source, ctrl+click reveals that this function isn't used anywhere in the code. I go digging around in the github, and I find this issue, and I find that it exists for some cryptic use-case that only exists in the Murnaghan job, which I'd argue is not relevant to most of the potential userbase for pyiron. Wow! 

Occupying this method name with code which exists for an extremely specific use-case for legacy puposes might have been acceptable when pyiron was for a niche userbase, but this is no longer acceptable if we're taking the viewpoint of general adoption.

Reply:
I just grepped `kill`  in both base and atomistics and actually nothing relevant pops up.  Either this went out quietly a long time ago or users were somehow expected to call it on the child jobs themselves? In both cases I'd say go ahead with the current direction.  If the "old" functionality is something we want to support again in master jobs, we can do that with cleaner code.